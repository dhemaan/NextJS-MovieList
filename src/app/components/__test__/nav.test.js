
import React from "react";
import { render} from "@testing-library/react";

import Nav from "@/app/components/Nav";


describe("Header Component", () => {
  
    it("renders the component correctly to match snapshot", async() => {
       const {container} = render(<Nav />);      
      expect(container).toMatchSnapshot();     
    });

    it("Verify No of links", async() => {
        const container = render(<Nav />);      
       const navLink = container.getByTestId("navLink")
       expect(navLink.childNodes.length).toEqual(4)     
     });

     it("Verify links name with their link", async() => {
        const container = render(<Nav />);      
       const navLink = container.getByTestId("navLink");
       
       expect(navLink.childNodes[0].childNodes[0].getAttribute("href")).toEqual("#")
       expect(navLink.childNodes[0].childNodes[0].text).toEqual("Home") 
       
       expect(navLink.childNodes[1].childNodes[0].getAttribute("href")).toEqual("/about")
       expect(navLink.childNodes[1].childNodes[0].text).toEqual("About") 
       
       expect(navLink.childNodes[2].childNodes[0].getAttribute("href")).toEqual("/movie")
       expect(navLink.childNodes[2].childNodes[0].text).toEqual("Movie") 
       
       expect(navLink.childNodes[3].childNodes[0].getAttribute("href")).toEqual("/contact")
       expect(navLink.childNodes[3].childNodes[0].text).toEqual("Contact") 
       
     });

  })
