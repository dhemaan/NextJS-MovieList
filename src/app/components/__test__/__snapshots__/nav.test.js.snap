// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders the component correctly to match snapshot 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class=""
    >
      <ul
        class="navbarList"
        data-testid="navLink"
      >
        <li
          class="navbarItem"
        >
          <a
            class="navbarLink"
            href="#"
          >
            Home
          </a>
        </li>
        <li
          class="navbarItem"
        >
          <a
            class="navbarLink"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="navbarItem"
        >
          <a
            class="navbarLink"
            href="/movie"
          >
            Movie
          </a>
        </li>
        <li
          class="navbarItem"
        >
          <a
            class="navbarLink"
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
      <div
        class="mobile-navbar-btn"
      >
        <svg
          class="mobile-nav-icon"
          fill="none"
          height="1em"
          name="menu-outline"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 6C2 5.44772 2.44772 5 3 5H21C21.5523 5 22 5.44772 22 6C22 6.55228 21.5523 7 21 7H3C2.44772 7 2 6.55228 2 6Z"
            fill="currentColor"
          />
          <path
            d="M2 12.0322C2 11.4799 2.44772 11.0322 3 11.0322H21C21.5523 11.0322 22 11.4799 22 12.0322C22 12.5845 21.5523 13.0322 21 13.0322H3C2.44772 13.0322 2 12.5845 2 12.0322Z"
            fill="currentColor"
          />
          <path
            d="M3 17.0645C2.44772 17.0645 2 17.5122 2 18.0645C2 18.6167 2.44772 19.0645 3 19.0645H21C21.5523 19.0645 22 18.6167 22 18.0645C22 17.5122 21.5523 17.0645 21 17.0645H3Z"
            fill="currentColor"
          />
        </svg>
        <svg
          class="mobile-nav-icon close-outline"
          fill="none"
          height="1em"
          name="close-outline"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.3956 7.75734C16.7862 8.14786 16.7862 8.78103 16.3956 9.17155L13.4142 12.153L16.0896 14.8284C16.4802 15.2189 16.4802 15.8521 16.0896 16.2426C15.6991 16.6331 15.0659 16.6331 14.6754 16.2426L12 13.5672L9.32458 16.2426C8.93405 16.6331 8.30089 16.6331 7.91036 16.2426C7.51984 15.8521 7.51984 15.2189 7.91036 14.8284L10.5858 12.153L7.60436 9.17155C7.21383 8.78103 7.21383 8.14786 7.60436 7.75734C7.99488 7.36681 8.62805 7.36681 9.01857 7.75734L12 10.7388L14.9814 7.75734C15.372 7.36681 16.0051 7.36681 16.3956 7.75734Z"
            fill="currentColor"
          />
          <path
            clip-rule="evenodd"
            d="M4 1C2.34315 1 1 2.34315 1 4V20C1 21.6569 2.34315 23 4 23H20C21.6569 23 23 21.6569 23 20V4C23 2.34315 21.6569 1 20 1H4ZM20 3H4C3.44772 3 3 3.44772 3 4V20C3 20.5523 3.44772 21 4 21H20C20.5523 21 21 20.5523 21 20V4C21 3.44772 20.5523 3 20 3Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </nav>
</div>
`;
